File: app/src/main/java/com/lumiyaviewer/lumiya/ui/settings/ModernSettingsActivity.java
package com.lumiyaviewer.lumiya.ui.settings;
import java.util.*;

import android.content.Intent;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.LinearLayout;
import android.widget.ScrollView;
import android.widget.Switch;
import android.widget.TextView;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

import com.lumiyaviewer.lumiya.LumiyaApp;

/**
 * Modern Settings Activity for Linkpoint Sample Application
 * Provides comprehensive configuration options for all modern components
 */
public class ModernSettingsActivity extends AppCompatActivity {
    private static final String TAG = "ModernSettingsActivity";
    
    // UI Components
    private Toolbar toolbar;
    private ScrollView scrollView;
    private LinearLayout mainLayout;
    
    // Settings sections
    private LinearLayout graphicsSection;
    private LinearLayout networkSection;
    private LinearLayout assetSection;
    private LinearLayout debugSection;
    
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        
        createSettingsLayout();
        setupToolbar();
        populateSettings();
    }
    
    private void createSettingsLayout() {
        // Create root scroll view
        scrollView = new ScrollView(this);
        scrollView.setFillViewport(true);
        
        // Create main layout
        mainLayout = new LinearLayout(this);
        mainLayout.setOrientation(LinearLayout.VERTICAL);
        mainLayout.setPadding(24, 16, 24, 24);
        
        // Create toolbar
        toolbar = new Toolbar(this);
        toolbar.setBackgroundColor(0xFF3F51B5);
        toolbar.setTitleTextColor(0xFFFFFFFF);
        LinearLayout.LayoutParams toolbarParams = new LinearLayout.LayoutParams(
            LinearLayout.LayoutParams.MATCH_PARENT, 
            (int) (56 * getResources().getDisplayMetrics().density));
        toolbar.setLayoutParams(toolbarParams);
        mainLayout.addView(toolbar);
        
        // Create settings sections
        createGraphicsSection();
        createNetworkSection();  
        createAssetSection();
        createDebugSection();
        createEmulatorSection();
        
        scrollView.addView(mainLayout);
        setContentView(scrollView);
    }
    
    private void setupToolbar() {
        setSupportActionBar(toolbar);
        if (getSupportActionBar() != null) {
            getSupportActionBar().setTitle("Demo Settings");
            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        }
    }
    
    private void createGraphicsSection() {
        graphicsSection = createSection("🎨 Graphics Settings");
        
        // Graphics quality setting
        addSwitchSetting(graphicsSection, 
                        "High Quality Graphics", 
                        "Enable maximum graphics quality (may impact battery)",
                        true);
        
        // PBR rendering setting
        addSwitchSetting(graphicsSection,
                        "PBR Rendering", 
                        "Use physically-based rendering (requires OpenGL ES 3.0+)",
                        true);
        
        // Texture compression setting
        addSwitchSetting(graphicsSection,
                        "Advanced Texture Compression", 
                        "Use ASTC/ETC2 compression for optimal quality",
                        true);
        
        // Frame rate limit
        addTextSetting(graphicsSection,
                      "Target Frame Rate", 
                      "Maximum FPS for graphics rendering",
                      "60 FPS");
        
        // GPU memory limit
        addTextSetting(graphicsSection,
                      "GPU Memory Limit", 
                      "Maximum GPU memory usage for textures",
                      "50 MB");
    }
    
    private void createNetworkSection() {
        networkSection = createSection("🌐 Network Settings");
        
        // Connection timeout
        addTextSetting(networkSection,
                      "Connection Timeout", 
                      "Timeout for network connections",
                      "30 seconds");
        
        // HTTP/2 setting
        addSwitchSetting(networkSection,
                        "Use HTTP/2 CAPS", 
                        "Enable modern HTTP/2 protocol for CAPS",
                        true);
        
        // WebSocket events
        addSwitchSetting(networkSection,
                        "WebSocket Events", 
                        "Use WebSocket for real-time event streaming",
                        true);
        
        // Connection pooling
        addSwitchSetting(networkSection,
                        "Connection Pooling", 
                        "Reuse connections for better performance",
                        true);
        
        // Bandwidth optimization
        addTextSetting(networkSection,
                      "Bandwidth Mode", 
                      "Network usage optimization level",
                      "Adaptive");
    }
    
    private void createAssetSection() {
        assetSection = createSection("📦 Asset Management");
        
        // Cache size
        addTextSetting(assetSection,
                      "Cache Size Limit", 
                      "Maximum disk space for asset cache",
                      "256 MB");
        
        // Quality level
        addTextSetting(assetSection,
                      "Default Quality Level", 
                      "Default asset quality for streaming",
                      "HIGH");
        
        // Preload setting
        addSwitchSetting(assetSection,
                        "Preload Critical Assets", 
                        "Download essential assets in advance",
                        true);
        
        // Compression setting
        addSwitchSetting(assetSection,
                        "Basis Universal Textures", 
                        "Use advanced texture transcoding",
                        true);
        
        // Background loading
        addSwitchSetting(assetSection,
                        "Background Asset Loading", 
                        "Continue downloading when app is backgrounded",
                        false);
    }
    
    private void createDebugSection() {
        debugSection = createSection("🔧 Debug & Testing");
        
        // Logging level
        addTextSetting(debugSection,
                      "Log Level", 
                      "Verbosity of debug logging",
                      "INFO");
        
        // Performance monitoring
        addSwitchSetting(debugSection,
                        "Performance Monitoring", 
                        "Track and report performance metrics",
                        true);
        
        // Network logging
        addSwitchSetting(debugSection,
                        "Network Request Logging", 
                        "Log all network requests and responses",
                        false);
        
        // Graphics debugging
        addSwitchSetting(debugSection,
                        "Graphics Debug Info", 
                        "Show graphics performance overlay",
                        false);
        
        // Memory usage tracking
        addSwitchSetting(debugSection,
                        "Memory Usage Tracking", 
                        "Monitor memory usage of components",
                        true);
        
        // Export logs button
        addActionButton(debugSection,
                       "📤 Export Debug Logs", 
                       "Export all application logs to Downloads",
                       this::exportDebugLogs);
        
        // Clear cache button
        addActionButton(debugSection,
                       "🗑️ Clear All Caches", 
                       "Clear asset cache and temporary data",
                       this::clearAllCaches);
        
        // Reset settings button
        addActionButton(debugSection,
                       "🔄 Reset to Defaults", 
                       "Reset all settings to default values",
                       this::resetToDefaults);
    }
    
    private LinearLayout createSection(String title) {
        // Section header
        TextView header = new TextView(this);
        header.setText(title);
        header.setTextSize(18);
        header.setTypeface(null, android.graphics.Typeface.BOLD);
        header.setPadding(16, 32, 16, 16);
        header.setTextColor(0xFF1976D2);
        mainLayout.addView(header);
        
        // Section container
        LinearLayout section = new LinearLayout(this);
        section.setOrientation(LinearLayout.VERTICAL);
        section.setPadding(16, 0, 16, 0);
        section.setBackgroundColor(0xFFF8F9FA);
        mainLayout.addView(section);
        
        return section;
    }
    
    private void addSwitchSetting(LinearLayout parent, String title, String description, boolean defaultValue) {
        // Container for this setting
        LinearLayout container = new LinearLayout(this);
        container.setOrientation(LinearLayout.HORIZONTAL);
        container.setPadding(16, 12, 16, 12);
        
        // Text container
        LinearLayout textContainer = new LinearLayout(this);
        textContainer.setOrientation(LinearLayout.VERTICAL);
        LinearLayout.LayoutParams textParams = new LinearLayout.LayoutParams(
            0, LinearLayout.LayoutParams.WRAP_CONTENT, 1.0f);
        textContainer.setLayoutParams(textParams);
        
        // Title
        TextView titleText = new TextView(this);
        titleText.setText(title);
        titleText.setTextSize(16);
        titleText.setTypeface(null, android.graphics.Typeface.BOLD);
        textContainer.addView(titleText);
        
        // Description
        TextView descText = new TextView(this);
        descText.setText(description);
        descText.setTextSize(12);
        descText.setTextColor(0xFF666666);
        textContainer.addView(descText);
        
        // Switch
        Switch switchView = new Switch(this);
        switchView.setChecked(defaultValue);
        
        container.addView(textContainer);
        container.addView(switchView);
        parent.addView(container);
    }
    
    private void addTextSetting(LinearLayout parent, String title, String description, String value) {
        // Container for this setting
        LinearLayout container = new LinearLayout(this);
        container.setOrientation(LinearLayout.HORIZONTAL);
        container.setPadding(16, 12, 16, 12);
        
        // Text container
        LinearLayout textContainer = new LinearLayout(this);
        textContainer.setOrientation(LinearLayout.VERTICAL);
        LinearLayout.LayoutParams textParams = new LinearLayout.LayoutParams(
            0, LinearLayout.LayoutParams.WRAP_CONTENT, 1.0f);
        textContainer.setLayoutParams(textParams);
        
        // Title
        TextView titleText = new TextView(this);
        titleText.setText(title);
        titleText.setTextSize(16);
        titleText.setTypeface(null, android.graphics.Typeface.BOLD);
        textContainer.addView(titleText);
        
        // Description
        TextView descText = new TextView(this);
        descText.setText(description);
        descText.setTextSize(12);
        descText.setTextColor(0xFF666666);
        textContainer.addView(descText);
        
        // Value
        TextView valueText = new TextView(this);
        valueText.setText(value);
        valueText.setTextSize(14);
        valueText.setTextColor(0xFF2196F3);
        valueText.setTypeface(null, android.graphics.Typeface.BOLD);
        
        container.addView(textContainer);
        container.addView(valueText);
        parent.addView(container);
    }
    
    private void addActionButton(LinearLayout parent, String title, String description, Runnable action) {
        // Container for this setting
        LinearLayout container = new LinearLayout(this);
        container.setOrientation(LinearLayout.VERTICAL);
        container.setPadding(16, 12, 16, 12);
        container.setClickable(true);
        container.setOnClickListener(v -> action.run());
        
        // Add slight background color for buttons
        container.setBackgroundColor(0xFFE3F2FD);
        
        // Title
        TextView titleText = new TextView(this);
        titleText.setText(title);
        titleText.setTextSize(16);
        titleText.setTypeface(null, android.graphics.Typeface.BOLD);
        titleText.setTextColor(0xFF1976D2);
        container.addView(titleText);
        
        // Description
        TextView descText = new TextView(this);
        descText.setText(description);
        descText.setTextSize(12);
        descText.setTextColor(0xFF666666);
        container.addView(descText);
        
        parent.addView(container);
    }
    
    private void populateSettings() {
        // This method would load settings from SharedPreferences
        // For the demo, we're using default values
    }
    
    private void exportDebugLogs() {
        // Simulate log export
        showToast("🔄 Exporting debug logs to Downloads...");
        
        // In a real implementation, this would export actual logs
        new Thread(() -> {
            try {
                Thread.sleep(2000); // Simulate export time
                runOnUiThread(() -> {
                    showToast("✅ Debug logs exported successfully!");
            } catch (InterruptedException e) {
                Thread.currentThread().interrupt();
            }
        }).start();
    }
    
    private void clearAllCaches() {
        // Simulate cache clearing
        showToast("🔄 Clearing all caches...");
        
        new Thread(() -> {
            try {
                Thread.sleep(3000); // Simulate clearing time
                runOnUiThread(() -> {
                    showToast("✅ All caches cleared successfully!");
            } catch (InterruptedException e) {
                Thread.currentThread().interrupt();
            }
        }).start();
    }
    
    private void resetToDefaults() {
        showToast("🔄 Resetting all settings to defaults...");
        
        new Thread(() -> {
            try {
                Thread.sleep(1000); // Simulate reset time
                runOnUiThread(() -> {
                    showToast("✅ Settings reset to default values!");
                    // In a real implementation, would reload the UI
            } catch (InterruptedException e) {
                Thread.currentThread().interrupt();
            }
        }).start();
    }
    
    private void showToast(String message) {
        android.widget.Toast.makeText(this, message, android.widget.Toast.LENGTH_SHORT).show();
    }
    
    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        menu.add(0, 1, 0, "Save Settings").setIcon(android.R.drawable.ic_menu_save);
        menu.add(0, 2, 0, "Help").setIcon(android.R.drawable.ic_menu_help);
        return true;
    }
    
    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                finish();
                return true;
            case 1: // Save
                saveSettings();
                return true;
            case 2: // Help
                showSettingsHelp();
                return true;
            default:
                return super.onOptionsItemSelected(item);
        }
    }
    
    private void createEmulatorSection() {
        LinearLayout emulatorSection = createSection("🔧 Emulator Management");
        
        // Emulator availability check
        EmulatorManager emulatorManager = new EmulatorManager(this);
        boolean isAvailable = emulatorManager.isAvailable();
        
        addTextSetting(emulatorSection,
                      "Emulator Tools Status", 
                      "Android emulator management availability",
                      isAvailable ? "Available" : "Not Available");
        
        // Quick emulator status
        addActionButton(emulatorSection,
                       "Check Emulator Status",
                       "Show running emulators and connected devices",
                       () -> {
                           if (isAvailable) {
                               emulatorManager.getStatus(new EmulatorManager.EmulatorCallback() {
                                   @Override
                                   public void onSuccess(String message) {
                                       runOnUiThread(() -> showToast("Status checked - see notification"));
                                   }
                                   
                                   @Override
                                   public void onError(String error) {
                                       runOnUiThread(() -> showToast("Status check failed"));
                                   }
                                   
                                   @Override
                                   public void onStatusUpdate(String status) {
                                       // Update UI if needed
                                   }
                           } else {
                               showToast("Emulator management not available");
                           }
        
        // Open full emulator settings
        addActionButton(emulatorSection,
                       "Open Emulator Settings",
                       "Access full emulator management interface",
                       () -> {
                           if (isAvailable) {
                               Intent intent = new Intent(this, EmulatorSettingsActivity.class);
                               startActivity(intent);
                           } else {
                               showToast("Emulator management not available");
                           }
        
        // Quick AVD list
        addActionButton(emulatorSection,
                       "List Available AVDs",
                       "Show currently configured Android Virtual Devices",
                       () -> {
                           if (isAvailable) {
                               emulatorManager.listAVDs(new EmulatorManager.EmulatorCallback() {
                                   @Override
                                   public void onSuccess(String output) {
                                       runOnUiThread(() -> {
                                           String summary = EmulatorManager.formatAVDSummary(
                                               EmulatorManager.parseAVDList(output));
                                           showToast("AVDs: " + summary.split("\n").length + " found");
                                   }
                                   
                                   @Override
                                   public void onError(String error) {
                                       runOnUiThread(() -> showToast("Failed to list AVDs"));
                                   }
                                   
                                   @Override
                                   public void onStatusUpdate(String status) {
                                       // Update UI if needed
                                   }
                           } else {
                               showToast("Emulator management not available");
                           }
    }
    
    private void saveSettings() {
        showToast("💾 Settings saved successfully!");
        // In a real implementation, would save to SharedPreferences
    }
    
    private void showSettingsHelp() {
        showToast("ℹ️ Settings help: Configure demo application behavior");
        // In a real implementation, would show detailed help dialog
    }
}

File: Linkpoint/src/main/kotlin/com/linkpoint/ui/settings/ModernSettingsActivity.kt
package com.linkpoint.ui.settings

import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.widget.LinearLayout
import android.widget.ScrollView
import android.widget.Switch
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import com.linkpoint.LinkpointApp

/**
 * Modern Settings Activity for Linkpoint Sample Application
 * Provides comprehensive configuration options for all modern components
 */
class ModernSettingsActivity : AppCompatActivity() {
    
    companion object {
        private const val TAG = "ModernSettingsActivity"
    }
    
    // UI Components
    private lateinit var toolbar: Toolbar
    private lateinit var scrollView: ScrollView
    private lateinit var mainLayout: LinearLayout
    
    // Settings sections
    private lateinit var graphicsSection: LinearLayout
    private lateinit var networkSection: LinearLayout
    private lateinit var assetSection: LinearLayout
    private lateinit var debugSection: LinearLayout
    
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        
        createSettingsLayout()
        setupToolbar()
        populateSettings()
    }
    
    private fun createSettingsLayout() {
        // Create root scroll view
        scrollView = ScrollView(this).apply {
            isFillViewport = true
        }
        
        // Create main layout
        mainLayout = LinearLayout(this).apply {
            orientation = LinearLayout.VERTICAL
            setPadding(24, 16, 24, 24)
        }
        
        // Create toolbar
        toolbar = Toolbar(this).apply {
            setBackgroundColor(0xFF3F51B5.toInt())
            setTitleTextColor(0xFFFFFFFF.toInt())
            layoutParams = LinearLayout.LayoutParams(
                LinearLayout.LayoutParams.MATCH_PARENT,
                (56 * resources.displayMetrics.density).toInt()
            )
        }
        mainLayout.addView(toolbar)
        
        // Create settings sections
        createGraphicsSection()
        createNetworkSection()  
        createAssetSection()
        createDebugSection()
        createEmulatorSection()
        
        scrollView.addView(mainLayout)
        setContentView(scrollView)
    }
    
    private fun setupToolbar() {
        setSupportActionBar(toolbar)
        supportActionBar?.apply {
            title = "Demo Settings"
            setDisplayHomeAsUpEnabled(true)
        }
    }
    
    private fun createGraphicsSection() {
        graphicsSection = createSection("🎨 Graphics Settings")
        
        // Graphics quality setting
        addSwitchSetting(
            graphicsSection, 
            "High Quality Graphics", 
            "Enable maximum graphics quality (may impact battery)",
            true
        )
        
        // PBR rendering setting
        addSwitchSetting(
            graphicsSection,
            "PBR Rendering", 
            "Use physically-based rendering (requires OpenGL ES 3.0+)",
            true
        )
        
        // Texture compression setting
        addSwitchSetting(
            graphicsSection,
            "Advanced Texture Compression", 
            "Use ASTC/ETC2 compression for optimal quality",
            true
        )
        
        // Frame rate limit
        addTextSetting(
            graphicsSection,
            "Target Frame Rate", 
            "Maximum FPS for graphics rendering",
            "60 FPS"
        )
        
        // GPU memory limit
        addTextSetting(
            graphicsSection,
            "GPU Memory Limit", 
            "Maximum GPU memory usage for textures",
            "50 MB"
        )
    }
    
    private fun createNetworkSection() {
        networkSection = createSection("🌐 Network Settings")
        
        // Connection timeout
        addTextSetting(
            networkSection,
            "Connection Timeout", 
            "Timeout for network connections",
            "30 seconds"
        )
        
        // HTTP/2 setting
        addSwitchSetting(
            networkSection,
            "Use HTTP/2 CAPS", 
            "Enable modern HTTP/2 protocol for CAPS",
            true
        )
        
        // WebSocket events
        addSwitchSetting(
            networkSection,
            "WebSocket Events", 
            "Use WebSocket for real-time event streaming",
            true
        )
        
        // Connection pooling
        addSwitchSetting(
            networkSection,
            "Connection Pooling", 
            "Reuse connections for better performance",
            true
        )
        
        // Bandwidth optimization
        addTextSetting(
            networkSection,
            "Bandwidth Mode", 
            "Network usage optimization level",
            "Adaptive"
        )
    }
    
    private fun createAssetSection() {
        assetSection = createSection("📦 Asset Management")
        
        // Cache size
        addTextSetting(assetSection,
                      "Cache Size Limit", 
                      "Maximum disk space for asset cache",
                      "256 MB")
        
        // Quality level
        addTextSetting(assetSection,
                      "Default Quality Level", 
                      "Default asset quality for streaming",
                      "HIGH")
        
        // Preload setting
        addSwitchSetting(assetSection,
                        "Preload Critical Assets", 
                        "Download essential assets in advance",
                        true)
        
        // Compression setting
        addSwitchSetting(assetSection,
                        "Basis Universal Textures", 
                        "Use advanced texture transcoding",
                        true)
        
        // Background loading
        addSwitchSetting(assetSection,
                        "Background Asset Loading", 
                        "Continue downloading when app is backgrounded",
                        false)
    }
    
    private fun createDebugSection() {
        debugSection = createSection("🔧 Debug & Testing")
        
        // Logging level
        addTextSetting(debugSection,
                      "Log Level", 
                      "Verbosity of debug logging",
                      "INFO")
        
        // Performance monitoring
        addSwitchSetting(debugSection,
                        "Performance Monitoring", 
                        "Track and report performance metrics",
                        true)
        
        // Network logging
        addSwitchSetting(debugSection,
                        "Network Request Logging", 
                        "Log all network requests and responses",
                        false)
        
        // Graphics debugging
        addSwitchSetting(debugSection,
                        "Graphics Debug Info", 
                        "Show graphics performance overlay",
                        false)
        
        // Memory usage tracking
        addSwitchSetting(debugSection,
                        "Memory Usage Tracking", 
                        "Monitor memory usage of components",
                        true)
        
        // Export logs button
        addActionButton(debugSection,
                       "📤 Export Debug Logs", 
                       "Export all application logs to Downloads",
                       this::exportDebugLogs)
        
        // Clear cache button
        addActionButton(debugSection,
                       "🗑️ Clear All Caches", 
                       "Clear asset cache and temporary data",
                       this::clearAllCaches)
        
        // Reset settings button
        addActionButton(debugSection,
                       "🔄 Reset to Defaults", 
                       "Reset all settings to default values",
                       this::resetToDefaults)
    }
    
    private fun createSection(title: String): LinearLayout {
        // Section header
        val header = TextView(this).apply {
            text = title
            textSize = 18f
            setTypeface(null, android.graphics.Typeface.BOLD)
            setPadding(16, 32, 16, 16)
            setTextColor(0xFF1976D2.toInt())
        }
        mainLayout.addView(header)
        
        // Section container
        val section = LinearLayout(this).apply {
            orientation = LinearLayout.VERTICAL
            setPadding(16, 0, 16, 0)
            setBackgroundColor(0xFFF8F9FA.toInt())
        }
        mainLayout.addView(section)
        
        return section
    }
    
    private fun addSwitchSetting(parent: LinearLayout, title: String, description: String, defaultValue: Boolean) {
        // Container for this setting
        val container = LinearLayout(this).apply {
            orientation = LinearLayout.HORIZONTAL
            setPadding(16, 12, 16, 12)
        }
        
        // Text container
        val textContainer = LinearLayout(this).apply {
            orientation = LinearLayout.VERTICAL
            layoutParams = LinearLayout.LayoutParams(
                0, LinearLayout.LayoutParams.WRAP_CONTENT, 1.0f
            )
        }
        
        // Title
        val titleText = TextView(this).apply {
            text = title
            textSize = 16f
            setTypeface(null, android.graphics.Typeface.BOLD)
        }
        textContainer.addView(titleText)
        
        // Description
        val descText = TextView(this).apply {
            text = description
            textSize = 12f
            setTextColor(0xFF666666.toInt())
        }
        textContainer.addView(descText)
        
        // Switch
        val switchView = Switch(this).apply {
            isChecked = defaultValue
        }
        
        container.addView(textContainer)
        container.addView(switchView)
        parent.addView(container)
    }
    
    private fun addTextSetting(parent: LinearLayout, title: String, description: String, value: String) {
        // Container for this setting
        val container = LinearLayout(this).apply {
            orientation = LinearLayout.HORIZONTAL
            setPadding(16, 12, 16, 12)
        }
        
        // Text container
        val textContainer = LinearLayout(this).apply {
            orientation = LinearLayout.VERTICAL
            layoutParams = LinearLayout.LayoutParams(
                0, LinearLayout.LayoutParams.WRAP_CONTENT, 1.0f
            )
        }
        
        // Title
        val titleText = TextView(this).apply {
            text = title
            textSize = 16f
            setTypeface(null, android.graphics.Typeface.BOLD)
        }
        textContainer.addView(titleText)
        
        // Description
        val descText = TextView(this).apply {
            text = description
            textSize = 12f
            setTextColor(0xFF666666.toInt())
        }
        textContainer.addView(descText)
        
        // Value
        val valueText = TextView(this).apply {
            text = value
            textSize = 14f
            setTextColor(0xFF2196F3.toInt())
            setTypeface(null, android.graphics.Typeface.BOLD)
        }
        
        container.addView(textContainer)
        container.addView(valueText)
        parent.addView(container)
    }
    
    private fun addActionButton(parent: LinearLayout, title: String, description: String, action: Runnable) {
        // Container for this setting
        val container = LinearLayout(this).apply {
            orientation = LinearLayout.VERTICAL
            setPadding(16, 12, 16, 12)
            isClickable = true
            setOnClickListener { action.run() }
            setBackgroundColor(0xFFE3F2FD.toInt())
        }
        
        // Title
        val titleText = TextView(this).apply {
            text = title
            textSize = 16f
            setTypeface(null, android.graphics.Typeface.BOLD)
            setTextColor(0xFF1976D2.toInt())
        }
        container.addView(titleText)
        
        // Description
        val descText = TextView(this).apply {
            text = description
            textSize = 12f
            setTextColor(0xFF666666.toInt())
        }
        container.addView(descText)
        
        parent.addView(container)
    }
    
    private fun populateSettings() {
        // This method would load settings from SharedPreferences
        // For the demo, we're using default values
    }
    
    private fun exportDebugLogs() {
        // Simulate log export
        showToast("🔄 Exporting debug logs to Downloads...")
        
        // In a real implementation, this would export actual logs
        Thread {
            try {
                Thread.sleep(2000) // Simulate export time
                runOnUiThread {
                    showToast("✅ Debug logs exported successfully!")
                }
            } catch (e: InterruptedException) {
                Thread.currentThread().interrupt()
            }
        }.start()
    }
    
    private fun clearAllCaches() {
        // Simulate cache clearing
        showToast("🔄 Clearing all caches...")
        
        Thread {
            try {
                Thread.sleep(3000) // Simulate clearing time
                runOnUiThread {
                    showToast("✅ All caches cleared successfully!")
                }
            } catch (e: InterruptedException) {
                Thread.currentThread().interrupt()
            }
        }.start()
    }
    
    private fun resetToDefaults() {
        showToast("🔄 Resetting all settings to defaults...")
        
        Thread {
            try {
                Thread.sleep(1000) // Simulate reset time
                runOnUiThread {
                    showToast("✅ Settings reset to default values!")
                    // In a real implementation, would reload the UI
                }
            } catch (e: InterruptedException) {
                Thread.currentThread().interrupt()
            }
        }.start()
    }
    
    private fun showToast(message: String) {
        android.widget.Toast.makeText(this, message, android.widget.Toast.LENGTH_SHORT).show()
    }
    
    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menu.add(0, 1, 0, "Save Settings").setIcon(android.R.drawable.ic_menu_save)
        menu.add(0, 2, 0, "Help").setIcon(android.R.drawable.ic_menu_help)
        return true
    }
    
    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            android.R.id.home -> {
                finish()
                true
            }
            1 -> { // Save
                saveSettings()
                true
            }
            2 -> { // Help
                showSettingsHelp()
                true
            }
            else -> super.onOptionsItemSelected(item)
        }
    }
    
    private fun createEmulatorSection() {
        val emulatorSection = createSection("🔧 Emulator Management")
        
        // Emulator availability check
        val emulatorManager = EmulatorManager(this)
        val isAvailable = emulatorManager.isAvailable()
        
        addTextSetting(
            emulatorSection,
            "Emulator Tools Status", 
            "Android emulator management availability",
            if (isAvailable) "Available" else "Not Available"
        )
        
        // Quick emulator status
        addActionButton(
            emulatorSection,
            "Check Emulator Status",
            "Show running emulators and connected devices",
            Runnable {
                if (isAvailable) {
                    emulatorManager.getStatus(object : EmulatorManager.EmulatorCallback {
                        override fun onSuccess(message: String) {
                            runOnUiThread { showToast("Status checked - see notification") }
                        }
                        
                        override fun onError(error: String) {
                            runOnUiThread { showToast("Status check failed") }
                        }
                        
                        override fun onStatusUpdate(status: String) {
                            // Update UI if needed
                        }
                    })
                } else {
                    showToast("Emulator management not available")
                }
            }
        )
        
        // Open full emulator settings
        addActionButton(
            emulatorSection,
            "Open Emulator Settings",
            "Access full emulator management interface",
            Runnable {
                if (isAvailable) {
                    val intent = Intent(this, EmulatorSettingsActivity::class.java)
                    startActivity(intent)
                } else {
                    showToast("Emulator management not available")
                }
            }
        )
        
        // Quick AVD list
        addActionButton(
            emulatorSection,
            "List Available AVDs",
            "Show currently configured Android Virtual Devices",
            Runnable {
                if (isAvailable) {
                    emulatorManager.listAVDs(object : EmulatorManager.EmulatorCallback {
                        override fun onSuccess(output: String) {
                            runOnUiThread {
                                val summary = EmulatorManager.formatAVDSummary(
                                    EmulatorManager.parseAVDList(output)
                                )
                                showToast("AVDs: ${summary.split("\n").size} found")
                            }
                        }
                        
                        override fun onError(error: String) {
                            runOnUiThread { showToast("Failed to list AVDs") }
                        }
                        
                        override fun onStatusUpdate(status: String) {
                            // Update UI if needed
                        }
                    })
                } else {
                    showToast("Emulator management not available")
                }
            }
        )
    }
    
    private fun saveSettings() {
        showToast("💾 Settings saved successfully!")
        // In a real implementation, would save to SharedPreferences
    }
    
    private fun showSettingsHelp() {
        showToast("ℹ️ Settings help: Configure demo application behavior")
        // In a real implementation, would show detailed help dialog
    }
}